### PR TITLE
feat: add list schema feature

### PR DESCRIPTION
**Pull request by eMerzh**
Maybe WIP... depends on what you want...

i'm trying to lays the bases for the schema support of sqlectron. 

So, for now This PR is only about listing schema :)
but i plan to do more soon... just getting some feedback (although this can be merged if you want)

i'm not sure the other supported DB support schema... i know that mysql doesn't but the other one... not really sure.

### CODE REVIEW

**Comment by eMerzh** [comment]
I've blindly added the support for sqlserver... not able to test it as the docker tests don't work for sqlserver and i don't have one myself.

File: src/db/client.js [hunk 1]
```diff
@@ -120,6 +121,11 @@ async function listTables(server, database) {
   return database.connection.listTables(database.database);
 }
 
+async function listSchemas(server, database) {
```
Comment (krolow) [inline src/db/client.js]: does not need async

File: src/db/clients/postgresql.js [hunk 2]
```diff
@@ -157,6 +158,22 @@ export function listTableTriggers(client, table) {
   });
 }
 
+export function listSchemas(client) {
+  return new Promise((resolve, reject) => {
+    const sql = `
+    SELECT schema_name
```
Comment (krolow) [inline src/db/clients/postgresql.js]: indention might be like the others https://github.com/eMerzh/sqlectron-core/blob/a88ac7b6981150fa5e67a7e414c13e5250b5185e/src/db/clients/postgresql.js#L102

File: src/db/client.js (hunk 1 above)
Comment (eMerzh) [inline src/db/client.js]: Ok i'll change that ... i did it because all other list*() have async :)
like listRoutines or listTableColumns

**Comment by krolow** [comment]
looks good @eMerzh just amend things in one commit, so we are ready to merge 👍

**Pull request merged**